src/core/parser.ml

```diff
@@ -308,6 +309,16 @@ let escaped_ident : bool -> string Earley.grammar = fun with_delim ->
 let escaped_ident_no_delim = escaped_ident false
 let escaped_ident = escaped_ident true
 
+(** [add_prefix p s] adds the prefix [p] at the beginning of the
+    string [s]. *)
+let add_prefix : string -> string -> string = fun p s ->
+  if Str.string_match (Str.regexp "^{|.*|}$") s 0 then
```

Comment:
No need for Str. Why not simply writing:
```
let n = String.length s in
if n >= 2 && s.[0] ='{' && s.[1] = '|' then "{|" ^ p ^ String.sub s 2 (n-4) ^ "|}"
else p ^ s
```

Reply:
`if n >= 2` is better?

Reply:
right